File: concrete/src/test/java/pcb/uwu/amount/composite/termodynamics/WattsTest.java
package pcb.uwu.amount.composite.termodynamics;

import org.junit.Test;
import pcb.uwu.amount.composite.electromagnetism.Volts;
import pcb.uwu.amount.quantity.Amperes;
import pcb.uwu.amount.quantity.Seconds;
import pcb.uwu.core.Magnitude;

import static java.math.MathContext.DECIMAL64;
import static org.junit.Assert.assertEquals;

public class WattsTest {
	private static final Watts watts = new Watts(20);

	@Test
	public void multipliedByTime() {
		assertEquals(new Joules(300),
				watts.multipliedBy(new Seconds(15), DECIMAL64));
	}

	@Test
	public void dividedByElectricCurrent() {
		assertEquals(new Volts(200),
				watts.dividedBy(new Amperes(100, Magnitude.MILLI), DECIMAL64));
	}

	@Test
	public void dividedByElectricPotential() {
		assertEquals(new Amperes(100, Magnitude.MILLI),
				watts.dividedBy(new Volts(200), DECIMAL64));
	}

}

File: implementation/src/test/java/pcb/uwu/amount/derived/thermodynamics/WattsTest.kt
package pcb.uwu.amount.derived.thermodynamics

import pcb.uwu.amount.base.Amperes
import pcb.uwu.amount.base.Seconds
import pcb.uwu.amount.derived.electromagnetism.Volts
import pcb.uwu.core.Magnitude.MILLI
import kotlin.test.Test
import kotlin.test.assertEquals

class WattsTest
{
    @Test
    fun multipliedByTime() =
        assertEquals(Joules(300),
                     Watts(20) * Seconds(15))

    @Test
    fun dividedByElectricCurrent() =
        assertEquals(Volts(200),
                     Watts(20) / Amperes(100, MILLI))

    @Test
    fun dividedByElectricPotential() =
        assertEquals(Amperes(0.1),
                     Watts(20) / Volts(200))
}
